Publish docker images to DockerHub

aryanpingle opened the pull request:
Intended to fix #783. 

Overview:
- Uses DockerHub as the container registry for the software's Docker images instead of GitHub Packages. (Note: the previously published GitHub packages will still show up under the `packages` tab of the sidebar)
- The DockerHub username must be a repository/environment variable called `DOCKERHUB_USERNAME`
- The DockerHub token must be a repository secret called `DOCKERHUB_TOKEN`
- Adds new markdown files to serve as DockerHub READMEs for the `client` and `libms` images

aryanpingle commented:
@prasadtalasila I created `servers/lib/DOCKER.md` to match the conventions of `client/DOCKER.md`, is there anything in it that should be removed/edited to make it suitable for the Docker Image?

aryanpingle commented on .github/workflows/docker.yml:
```diff
@@ -3,10 +3,10 @@ name: Build and Push Docker Image
-        required: false
-        type: string
-        default: "ghcr.io"
+      # registry:
+      #   required: false
+      #   type: string
+      #   default: "ghcr.io"
```
Removed this because we'll be using the default DockerHub registry.

prasadtalasila replied:
We need to publish on both github nd docker hub.

aryanpingle commented:
Additionally, many files reference the Docker Images on GitHub packages. Should we remove/edit those sections?

prasadtalasila commented:
@aryanpingle we need to publish to both github and docker hub. Does the PR already support it?

aryanpingle commented:
Ah no, only to DockerHub. I'll add GH package support back.

aryanpingle commented:
@prasadtalasila I've added back the GitHub Packages functionality.

Some notes:

- I've split the GitHub Packages and DockerHub workflows into separate files, and called both in separate jobs. I feel that calling both separately offers more control over the workflows, but I could put them in a combined `docker.yml` file and simply call that instead.
- The `check_version` step incorrectly left out the user name, which would result in it always assuming the version didn't exist. I've included the fix for it in [this commit](https://github.com/INTO-CPS-Association/DTaaS/pull/958/commits/f4e557cef85bcb66c68df25eaf32d65d3b2b76e3).
- In `lib-ms.yml` the `publish-docker-image` job did not include the push+branch condition, was this intentional?

prasadtalasila commented:
@aryanpingle 
Please use these two edited files.
[Client-DOCKER.md](https://github.com/user-attachments/files/17310371/Client-DOCKER.md)
[LIB-DOCKER.md](https://github.com/user-attachments/files/17310374/LIB-DOCKER.md)

aryanpingle commented on .github/workflows/docker-dockerhub.yml:
```diff
@@ -0,0 +1,78 @@
+        required: true
+        type: string
+    secrets:
+      DOCKERHUB_TOKEN:
```
Made it a repository secret instead
